lib/selection/service/application_distribution.rb

```diff
@@ -0,0 +1,33 @@
+module Selection
+  module Service
+    class ApplicationDistribution
+      def initialize(application:)
+        @application = application
+      end
+
+      def distribute
+        reviewers.each do |reviewer|
+          Todo.create!(user: reviewer, application: @application)
+        end
+      end
+
+      private
+
+      def assigned_reviewers
+        Todo.where(application_id: @application.id).collect(&:user)
+      end
+
+      def reviewers
+        all_reviewers = User.with_role("reviewer").to_a
+
+        # remove already assigned reviewers
+        possible_reviewers = all_reviewers - assigned_reviewers
+
+        # sort by assigned applications to distribute the work evenly
+        sorted_reviewers = possible_reviewers.shuffle.sort_by { |u| u.todos.count }
+
```

Comment:
@F3PiX good question. since we only give sort the todos count, it does not sort the users by itself.

I did a quick example to show it a little better. lets represent the users as a little hash with user id and the already assigned applications.

```
users = [
  { user_id: 1, assigned_applications: 0 },
  { user_id: 2, assigned_applications: 2 },
  { user_id: 3, assigned_applications: 0 },
  { user_id: 4, assigned_applications: 0 },
  { user_id: 5, assigned_applications: 1 },
  { user_id: 6, assigned_applications: 0 },
  { user_id: 7, assigned_applications: 1 },
  { user_id: 8, assigned_applications: 1 },
  { user_id: 9, assigned_applications: 2 },
  { user_id: 10, assigned_applications: 0 },
  { user_id: 11, assigned_applications: 1 },
  { user_id: 12, assigned_applications: 2 },
  { user_id: 13, assigned_applications: 3 },
  { user_id: 14, assigned_applications: 5 },
  { user_id: 15, assigned_applications: 0 }
]
```

after shuffling it looks like this

```
users.shuffle!

[ {:user_id=>10, :assigned_applications=>0},
  {:user_id=>6, :assigned_applications=>0},
  {:user_id=>1, :assigned_applications=>0},
  {:user_id=>12, :assigned_applications=>2},
  {:user_id=>14, :assigned_applications=>5},
  {:user_id=>8, :assigned_applications=>1},
  {:user_id=>13, :assigned_applications=>3},
  {:user_id=>2, :assigned_applications=>2},
  {:user_id=>5, :assigned_applications=>1},
  {:user_id=>4, :assigned_applications=>0},
  {:user_id=>15, :assigned_applications=>0},
  {:user_id=>11, :assigned_applications=>1},
  {:user_id=>3, :assigned_applications=>0},
  {:user_id=>9, :assigned_applications=>2},
  {:user_id=>7, :assigned_applications=>1}
]
```

and after sorting
```
users.sort_by { |u| u[:assigned_applications] }


[ {:user_id=>15, :assigned_applications=>0},
  {:user_id=>4, :assigned_applications=>0},
  {:user_id=>1, :assigned_applications=>0},
  {:user_id=>10, :assigned_applications=>0},
  {:user_id=>3, :assigned_applications=>0},
  {:user_id=>6, :assigned_applications=>0},
  {:user_id=>7, :assigned_applications=>1},
  {:user_id=>8, :assigned_applications=>1},
  {:user_id=>5, :assigned_applications=>1},
  {:user_id=>11, :assigned_applications=>1},
  {:user_id=>12, :assigned_applications=>2},
  {:user_id=>9, :assigned_applications=>2},
  {:user_id=>2, :assigned_applications=>2},
  {:user_id=>13, :assigned_applications=>3},
  {:user_id=>14, :assigned_applications=>5}
]
```

since we want to evenly distribute we somehow need to pick the users with less assigned applications first.